tensordict/tensordict.py

```diff
@@ -1938,10 +1941,16 @@ def set_default(
         return self.get(key)
 
     def lock(self):
-        self.is_locked = True
+        self._is_locked = True
```

Comment:
This looks cool! Maybe I missed this in the docs, but what does locking do ?

Reply:
It's not well documented :( 
The idea is to prevent writing new tensors in a tensordict.
If you put your TD in shared memory, we don't have a mechanism to update the tensors it contains in every process. Hence we lock it, such that the keys are fixed and can't be changed.
You can update it in-place though, simply not add or remove keys.

Reply:
Got it - thanks for the clarification :)